src/main/java/updater/JarReplaceAndExecute.java

```diff
@@ -0,0 +1,193 @@
+            log("Such source file does not exist, aborting.");
+            quit();
+        }
+
+        // make backup if target file exists
+        if (targetJarFile.exists() && !makeBackup(targetJarFile)) {
+            log("Failed to make backup.");
+            quit();
+        }
+
+        // move source to target
+        log("Moving source to target");
+        if (!moveFile(sourceJarFile.toPath(), targetJarFile.toPath())) {
+            log("Failed to move update.");
+            restoreBackup(targetJarFile);
+            quit();
+        }
+
+        // start target
+        // - if target can't be started, rollback backup
+        if (!executeJar(targetJarPath, "")) {
+            restoreBackup(targetJarFile);
+            executeJar(targetJarPath, "");
+        } else {
+            removeBackup(targetJarFile);
+        }
+
+        quit();
+    }
+
+    private boolean moveFile(Path source, Path dest) {
+        log("Moving file from:" + source.toString() + " to:" + dest.toString());
+        try {
+            if (!dest.toFile().exists() && !dest.toFile().createNewFile()) {
+                return false;
+            }
+            Files.move(source, dest, StandardCopyOption.REPLACE_EXISTING);
+        } catch (IOException e) {
+            log("Failed to move file");
+            log(e.toString());
+            return false;
+        }
+
+        return true;
+    }
+
+    public void quit() {
+        Platform.exit();
+        System.exit(0);
+    }
+
+    private boolean makeBackup(File file) {
+        log("Making backup");
+        String backupFilename = getBackupFilename(file.getName());
+        File backup = new File(file.getParent(), backupFilename);
+        return moveFile(file.toPath(), backup.toPath());
+    }
+
+    private boolean restoreBackup(File file) {
+        log("Restoring backup");
+        String backupFilename = getBackupFilename(file.getName());
+        File backup = new File(file.getParent(), backupFilename);
+        return moveFile(backup.toPath(), file.toPath());
+    }
+
+    private void removeBackup(File file) {
+        log("Removing backup");
+        String backupFilename = getBackupFilename(file.getName());
+        File backup = new File(file.getParent(), backupFilename);
+        if (!backup.delete()) {
+            log("Failed to remove backup");
+        }
+    }
+
+    private String getBackupFilename(String filename) {
+        return filename + BACKUP_FILENAME_APPENDIX;
+    }
+
+    /**
+     * Runs a command to execute a JAR file with the given arguments and options
+     * @param jarPath path (in string) to JAR file to be executed
+     * @param argsAndOptions raw arguments and options to run the JAR
+     * @return true if JAR can be executed, false otherwise
+     */
+    private boolean executeJar(String jarPath, String argsAndOptions) {
+        String command = "java -jar " + jarPath + " " + argsAndOptions;
+        command = command.trim();
```

Comment:
Can this method be static?

This line would be a bit nicer to read with String.format(...).trim().